### PR TITLE
Jczaja/xpu doc

### PR DESCRIPTION
**Pull request by jczaja**
Documentation updated with section on using mpi4jax with Intel XPU

### CODE REVIEW

File: docs/installation.rst [hunk 1]
```diff
@@ -86,3 +86,30 @@ If this is a bottleneck in your application, you can build MPI with CUDA support
...
+
+Installation with Intel GPU/XPU support
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+``mpi4jax`` also supports JAX arrays stored in Intel GPU/XPU memory.
+
+**Note:** To use mpi4jax and Jax on the Intel XPU or Intel GPU you need
+to have installed
+`intel-extension-for-openxla <https://github.com/intel/intel-extension-for-openxla>`__
+at least in version 0.2.1
```
Comment (Silv3S) [inline docs/installation.rst]: **0.2.1** is not sufficient version

File: docs/installation.rst [hunk 2]
```diff
@@ -86,3 +86,30 @@ If this is a bottleneck in your application, you can build MPI with CUDA support
...
+
+To build ``mpi4jax``'s XPU extensions, we need to locate SYCL headers
+and libraries on your system. SYCL comes as part of `Intel BaseKit
+AI <https://www.intel.com/content/www/us/en/developer/tools/oneapi/ai-analytics-toolkit.html>`__.
```
Comment (Silv3S) [inline docs/installation.rst]:
```suggestion
To build ``mpi4jax``'s XPU extensions, we need to locate SYCL headers
and libraries on your system. SYCL comes as part of `Intel oneAPI Base Toolkit <https://www.intel.com/content/www/us/en/developer/tools/oneapi/ai-analytics-toolkit.html>`__.
```

**Review by sfraczek — COMMENTED** [review]
The doc points to instruction in yml file where mpich is installed
```yml
- name: Setup MPI (mpich)
      uses: mpi4py/setup-mpi@v1
      with:
        mpi: mpich
```
Then sources basekit and also exports ONEAPI_ROOT which should be handled by sourcing and doesn't add basekit.config with `exclude=mpi`. I wonder which mpi is used there.
```yml
    - name: Build XPU extensions
      run: |
        source $HOME/basekit/setvars.sh
        export ONEAPI_ROOT=$HOME/basekit
        python setup.py build_ext --inplace
        test -f mpi4jax/_src/xla_bridge/mpi_xla_bridge_xpu*.so
```

**Comment by Silv3S** [comment]
> The doc points to instruction in yml file where mpich is installed
> 
> ```yaml
> - name: Setup MPI (mpich)
>       uses: mpi4py/setup-mpi@v1
>       with:
>         mpi: mpich
> ```
> 
> Then sources basekit and also exports ONEAPI_ROOT which should be handled by sourcing and doesn't add basekit.config with `exclude=mpi`. I wonder which mpi is used there.
> 
> ```yaml
>     - name: Build XPU extensions
>       run: |
>         source $HOME/basekit/setvars.sh
>         export ONEAPI_ROOT=$HOME/basekit
>         python setup.py build_ext --inplace
>         test -f mpi4jax/_src/xla_bridge/mpi_xla_bridge_xpu*.so
> ```

This action downloads just [SYCL Compiler](https://registrationcenter-download.intel.com/akdlm/IRC_NAS/bb99984f-370f-413d-bbec-38928d2458f2/l_dpcpp-cpp-compiler_p_2024.0.2.29_offline.sh), not whole basekit. Here I paste logs from calling setvars.sh:
```
:: initializing oneAPI environment ...
   85fb0d[12](https://github.com/Silv3S/mpi4jax/actions/runs/7887042164/job/21521438170#step:8:13)-e47b-4f2c-b95b-afc2078ef300.sh: BASH_VERSION = 5.1.[16](https://github.com/Silv3S/mpi4jax/actions/runs/7887042164/job/21521438170#step:8:17)(1)-release
   args: Using "$@" for setvars.sh arguments: 
:: compiler -- latest
:: debugger -- latest
:: dev-utilities -- latest
:: dpl -- latest
:: tbb -- latest
:: oneAPI environment initialized ::
```

File: docs/installation.rst (hunk 1 above)
Comment (jczaja) [reply docs/installation.rst]: ok, I will bump up to 0.3.0

File: docs/installation.rst (hunk 2 above)
Comment (jczaja) [reply docs/installation.rst]: ok

**Comment by sfraczek** [comment]
@Silv3S after setvars.sh you don't already have ONEAPI_ROOT exported?

**Comment by Silv3S** [comment]
> @Silv3S after setvars.sh you don't already have ONEAPI_ROOT exported?

You're right, exporting ONEAPI_ROOT can be deleted as setvars.sh already sets ONEAPI_ROOT ([logs from test](https://github.com/Silv3S/mpi4jax/actions/runs/7990968852/job/21821010279#step:8:26))

**Review by sfraczek — APPROVED** [review]
LGTM

**Pull request merged**